### PR TITLE
build: env variables improvements

### PR DESCRIPTION
**Pull request by sebastianmusial**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

* We had to set up the app URL in multiple places 
* hard configuration for multiple instances

## What is the new behavior?

* added new env variable (.env) - `APP_URL` which represents the main app URL for the iframe

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: libs/ai-embedded/set-env.mjs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    const envKey = variableMap[key];
+    const value = process.env[envKey];
+    if (value) {
+      const regex = new RegExp(`(${key}:\\s*).*(,)`, 'g');
+      content = content.replace(regex, `$1'${value}'$2`);
+    }
+  });
+
+  writeFileSync(filePath, content, 'utf8');
+  console.log(`Updated ${filePath}`);
```
Comment (olivierhalupczok) [inline libs/ai-embedded/set-env.mjs:26]: There is a chance for no updates to be made. Maybe we would like to inform the user about the update, only when there was one?

Comment (olivierhalupczok) [reply libs/ai-embedded/set-env.mjs:26]:
```ts
  if (updatedContent !== content) {
    writeFileSync(filePath, updatedContent, 'utf8');
    console.log(`Updated ${filePath}`);
  } else {
    console.log(`No changes made to ${filePath}`);
  }
```

Comment (olivierhalupczok) [reply libs/ai-embedded/set-env.mjs:26]: Or without the logic included in the `else`

Comment (sebastianmusial) [reply libs/ai-embedded/set-env.mjs:26]: It is a script for a prebuild process. Variables are always replaced based on the map (`variableMap`).  I've implemented a regex there and I don't compare any values (old - new).  If we would like to do it like you mentioned then we will need to check if we change sth in any variable in the map. I am not sure if we need to complicate it 😅

**Review by olivierhalupczok — APPROVED** [review]

**Pull request merged**